.github/workflows/deletedFiles.yml

```diff
@@ -18,18 +21,34 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Update deleted files
+      - name: Cleaning list from re-introduced files
+        run: |
+          for E in $(git ls-tree -r master --name-only); do
+            if (git check-attr export-ignore "$E" | grep -q "export-ignore: set"); then
+              continue
+            fi
+            grep -v "^$E$" data/deleted.files > data/deleted.files.tmp && mv data/deleted.files{.tmp,}
+          done
+          
+      - name: Update list with deleted files
         run: |
           for F in $(git diff origin/stable..HEAD --summary | awk '/^ delete/ && $4 !~ /^(VERSION)/ {print $4}'); do
             if (git check-attr export-ignore "$F" | grep -q "export-ignore: set"); then
               continue
             fi
-            if grep -q "^$F" data/deleted.files; then
+            if grep -q "^$F$" data/deleted.files; then
               continue
             fi
-            echo "$F" >> data/deleted.files
+            if ( ! test -f "data/deleted.files.tmp"); then
+              awk -v "input=# newly removed" '/# removed in/ && !found {print input; found=1} 1' data/deleted.files > data/deleted.files.tmp && cp data/deleted.files{.tmp,}
+            fi
+            awk -v "input=$F" '/# removed in/ && !found {print input; found=1} 1' data/deleted.files > data/deleted.files.tmp && cp data/deleted.files{.tmp,}
           done
-
+          if (test -f "data/deleted.files.tmp"); then
+            awk '/# removed in/ && !found {printf("\n"); found=1} 1' data/deleted.files > data/deleted.files.tmp && mv data/deleted.files{.tmp,}
+            sed -i "s/^# newly removed/# removed in $(date -I)/" data/deleted.files
```

Comment:
this would resolve #3949

Reply:
Incorrect preposition, should be "on (date)" => #4263

Reply:
Point taken, however the wording was taken from the original code and the meaning is "removed in <release>".

For the time being, one can presume the release tag like e.g., 2024-02-06a for Kaos will be revised manually in a final stage for the master branch.  For the future, one could furtherly improve this workflow by automatically collecting all new remove dates (if there are any and more than one date entry given after the last release) in the last steps of providing a release and even receive the applicable release tag from git action.

But first "Let's see how it will work in real life..." and second maybe worth a feature request?